File: src/main/resources/webapp/webapp/src/app/shared/offer.ts
export class Offer {


  constructor(public id: String,
              public _title: String,
              public _description: String,
              public _category: String) {
  }


  public get title(): String {
    return this._title;
  }

  public set title(value: String) {
    this._title = value;
  }

  public get description(): String {
    return this._description;
  }

  public set description(value: String) {
    this._description = value;
  }

  public get category(): String {
    return this._category;
  }

  public set category(value: String) {
    this._category = value;
  }
}

File: server/src/main/java/com/workoffice/entity/Offer.java
package com.workoffice.entity;

import javax.persistence.Id;


public class Offer {

    @Id
    private String id;
    private String title;
    private String description;
    private String category;
    private int userId;

    public Offer() {
    }

    public Offer(String title, String description, String category, int userId) {
        this.title = title;
        this.description = description;
        this.category = category;
        this.userId = userId;
    }

    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getCategory() {
        return category;
    }

    public int getUserId() {
        return userId;
    }

    @Override
    public String toString() {
        return "Offer{" +
                "id='" + id + '\'' +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", category='" + category + '\'' +
                ", userId=" + userId +
                '}';
    }
}
